test_libs/pyspec/eth2spec/test/context.py

```diff
@@ -80,3 +88,35 @@ def entry(*args, **kw):
         bls.bls_active = old_state
         return out
     return entry
+
+
+def with_phase0(fn):
+    """
+    Decorator to use phase 0's spec and helpers
+    """
+    def entry(*args, **kw):
+        kw['spec'] = spec_phase0
+        return fn(*args, **kw)
+    return entry
+
+
+def with_phase1(fn):
+    """
+    Decorator to use phase 1's spec and helpers
+    """
+    def entry(*args, **kw):
+        kw['spec'] = spec_phase1
+        return fn(*args, **kw)
+    return entry
+
+
+def with_all_phases(fn):
+    """
+    Decorator to run everything with all availible spec phases
+    """
+    def entry(*args, **kw):
+        kw['spec'] = spec_phase0
+        fn(*args, **kw)
+        kw['spec'] = spec_phase1
+        return fn(*args, **kw)
+    return entry
```

Comment:
Basically, I like `with_all_phases` approach!

Some issues:
1. The state generator is broken now. I run:
```sh
make eth2.0-spec-tests/tests/sanity
```
(Not sure if I missed any new flag?)

Error message:
```sh
Traceback (most recent call last):
  File "main.py", line 34, in <module>
    create_suite('slots', 'mainnet', lambda: generate_from_tests(test_slots)),
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/gen_base/gen_runner.py", line 98, in run_generator
    (output_name, handler, suite) = suite_creator(args.configs_path)
  File "main.py", line 25, in suite_definition
    test_cases=get_cases()))
  File "main.py", line 31, in <lambda>
    create_suite('blocks', 'minimal', lambda: generate_from_tests(test_blocks)),
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/gen_from_tests/gen.py", line 19, in generate_from_tests
    test_case = tfn(generator_mode=True, bls_active=bls_active)
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/eth2spec/test/context.py", line 121, in entry
    return fn(*args, **kw)
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/eth2spec/test/context.py", line 13, in entry
    kw['state'] = create_genesis_state(spec=kw['spec'], num_validators=spec_phase0.SLOTS_PER_EPOCH * 8)
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/eth2spec/test/helpers/genesis.py", line 35, in create_genesis_state
    state.validator_registry = [build_mock_validator(spec, i, state.balances[i]) for i in range(num_validators)]
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/eth2spec/test/helpers/genesis.py", line 35, in <listcomp>
    state.validator_registry = [build_mock_validator(spec, i, state.balances[i]) for i in range(num_validators)]
  File "/Users/hwwang/ethereum/eth2.0-specs/test_generators/sanity/venv/lib/python3.6/site-packages/eth2spec/test/helpers/genesis.py", line 8, in build_mock_validator
    withdrawal_credentials = spec.BLS_WITHDRAWAL_PREFIX_BYTE + spec.hash(pubkey)[1:]
AttributeError: module 'eth2spec.phase1.spec' has no attribute 'BLS_WITHDRAWAL_PREFIX_BYTE'
generator /sanity finished
```

2. It looks like the state test generator output would be only the result of `spec_phase1`, no `spec_phase0`?